opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -520,7 +520,7 @@ namespace Opm
 
         // store a copy of the well state, we don't want to update the real well state
         WellState<Scalar> well_state_copy = well_state;
-        const auto& group_state = simulator.problem().wellModel().groupState();
+        auto group_state = simulator.problem().wellModel().groupState();
```

Comment:
you might want to use `group_state_copy` to let people know it is a copy.  Same with `StandardWell_impl.hpp`.